### DISCUSSION TITLE
C# and WASM modules with imports and exports

### DISCUSSION THREAD

**Discussion by ero-qt** [dotnet/runtime#104462] [Q&A]
Hello everyone.

An app I'm using is able to load WASM modules/components. I would like to write this module in C#. For this, I need to export a function named `update` that the app calls, and I need to import some functions that the app provides. An entry point is not necessary.

Is this possible in C# as of today? How can I create this? What things do I need to keep in mind? Are there any techniques to speed up performance or lower output size?

**Comment by maraf** [comment]
Similar to https://github.com/dotnet/runtime/discussions/102427. Does it answer your question completely?
